src/context-parser-handlebars.js

```diff
@@ -401,8 +413,54 @@ ContextParserHandlebars.prototype.analyzeAst = function(ast, contextParser, char
 
                 output += t.output;
 
+            } else if (node.type === handlebarsUtils.PARTIAL_EXPRESSION) {
+
+                // if the 'partial expression' is not in Data State,
+                // TODO: we support basic partial only, need to enhance it.
+                // http://handlebarsjs.com/partials.html
+
+                if (parser.getCurrentState() !== stateMachine.State.STATE_DATA) {
+                    var re = handlebarsUtils.isValidExpression(node.content, 0, handlebarsUtils.PARTIAL_EXPRESSION),
+                        partialName, enterState, partialOutput, partialContent;
+                    if (re && this._partialsCache[re.tag]) {
+                        partialName = re.tag;
+                        partialContent = this._partialsCache[partialName];
+                        var partialParser = new ContextParserHandlebars({
+                                                     printCharEnable: false,
+                                                     enablePartialCombine: this._config._enablePartialCombine,
+                                                     strictMode: this._config._strictMode});
+
+                        // clone the state of the current context parser
+                        enterState = parser.getLastState();
```

Comment:
seems like [getLastState()](https://github.com/yahoo/context-parser/blob/master/src/context-parser.js#L570) is a duplicated of [getCurrentState()](https://github.com/yahoo/context-parser/blob/master/src/context-parser.js#L623)

c.c. @yukinying